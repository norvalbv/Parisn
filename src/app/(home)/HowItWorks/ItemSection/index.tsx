import { ReactElement } from 'react';
import StyledLink from '@/src/components/StyledLink';

type ItemSectionProps = {
  imageSrc: string;
  imageAlt: string;
  title: string;
  description: string;
  isReversed?: boolean;
  showLink?: boolean;
};

const ItemSection = ({
  imageSrc,
  imageAlt,
  title,
  description,
  isReversed = false,
  showLink = false,
}: ItemSectionProps): ReactElement => {
  return (
    <div className="flex h-screen items-center">
      <div className="mx-auto flex w-full max-w-7xl flex-col items-center gap-8 px-4 sm:gap-4 sm:p-8 md:w-4/5 md:flex-row md:p-20">
        <div className={`flex flex-1 justify-center ${isReversed ? 'md:order-2' : ''}`}>
          <img
            src={imageSrc}
            alt={imageAlt}
            className="w-[250px] rounded-xl object-cover sm:w-[300px] md:h-[400px] md:w-[400px]"
          />
        </div>
        <div
          className={`flex-1 px-4 text-center sm:px-0 md:text-left ${
            isReversed ? 'md:order-1' : ''
          }`}
        >
          <h4 className="text-xl font-semibold uppercase tracking-wide sm:text-2xl">{title}</h4>
          <p className="mb-4 mt-2 w-full text-sm text-text-secondary sm:text-base md:mb-6 md:w-4/5">
            {description}
          </p>
          {showLink && <StyledLink href="/how-it-works#step-three" />}
        </div>
      </div>
    </div>
  );
};

export default ItemSection;
