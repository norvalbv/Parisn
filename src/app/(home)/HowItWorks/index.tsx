'use client';

import React, { ReactElement, useRef } from 'react';
import StyledLink from '@/src/components/StyledLink';
import PRODUCT_1_IMAGE from '@/public/images/shoe.jpg';
import { motion, useScroll, useTransform } from 'framer-motion';
import WavyBackground from '@/src/components/Waves';

const MAX_PRICE = 1000;
const MIN_PRICE = 0;

const HowItWorks = (): ReactElement => {
  const containerRef = useRef<HTMLDivElement>(null);
  const priceRef = useRef<HTMLDivElement>(null);

  const { scrollYProgress } = useScroll({
    target: priceRef,
    offset: ['start end', 'end start'],
  });

  const rawPrice = useTransform(
    scrollYProgress,
    [0.2, 0.8],
    [MAX_PRICE, MIN_PRICE]
  );

  const price = useTransform(rawPrice, (value: number) => value.toFixed(2));

  return (
    <div className="relative min-h-[300vh]">
      <div className="sticky top-0 h-screen w-full overflow-hidden">
        <WavyBackground
          colors={[
            'rgba(50, 117, 248, 0.15)', 
            'rgba(255, 192, 203, 0.15)',
            'rgba(255, 255, 255, 0.1)',
          ]}
          backgroundFill="transparent"
          speed="slow"
          waveOpacity={0.3}
        />
      </div>

      <section ref={containerRef} className="absolute inset-0 py-10 sm:py-20">
        <div className="flex h-screen items-center">
          <div className="mx-auto flex w-full max-w-7xl flex-col items-center gap-8 px-4 sm:gap-4 sm:p-8 md:w-4/5 md:flex-row md:p-20">
            <img
              src={PRODUCT_1_IMAGE.src}
              alt="Luxury product showcase"
              className="w-[250px] rounded-xl object-cover sm:w-[300px] md:h-[400px] md:w-[400px]"
            />
            <div className="px-4 text-center sm:px-0 md:text-left">
              <h4 className="text-xl font-semibold uppercase tracking-wide sm:text-2xl">
                Discover Your Perfect Piece
              </h4>
              <p className="mb-4 mt-2 w-full text-sm leading-relaxed text-primary-neutral/80 sm:text-base md:mb-6 md:w-4/5">
                Explore our curated collection of luxury items. Hand curated by our team of experts,
                uniquely yours.
              </p>
            </div>
          </div>
        </div>

        <motion.div ref={priceRef} className="flex min-h-[150vh] md:min-h-dvh items-center"
            initial={{ opacity: 0, y: -20 }}
            animate={{ opacity: 1, y: 0 }}
            exit={{ opacity: 0, y: 20 }}
            transition={{ duration: 0.5 }}>
          <div className="sticky top-1/2 w-full -translate-y-1/2">
            <div className="mx-auto flex w-full max-w-7xl flex-col items-center justify-between gap-8 p-4 sm:gap-16 sm:p-8 md:flex-row md:p-20">
              <div className="w-full space-y-4 text-center sm:space-y-6 md:w-1/3 md:text-left">
                <h4 className="text-xl font-semibold uppercase tracking-wide sm:text-2xl">
                  Experience Dynamic Pricing
                </h4>
                <p className="mb-4 mt-2 w-full text-sm leading-relaxed text-primary-neutral/80 sm:text-base md:mb-6">
                  <span>Watch prices decrease in real-time. Our innovative system rewards patience,
                  but remember - popular items may sell quickly at higher prices.</span>
                </p>
              </div>

              <div className="relative flex w-full justify-center md:w-2/3 md:justify-end">
                <div className="group relative flex h-[120px] w-[300px] items-center justify-start overflow-hidden rounded-3xl border border-white/20 bg-gradient-to-br from-zinc-900/90 via-zinc-900/80 to-zinc-800/60 px-8 shadow-[0_0_50px_rgba(0,0,0,0.3)] backdrop-blur-xl transition-all duration-300 hover:border-white/30 hover:shadow-[0_0_80px_rgba(0,0,0,0.4)] sm:h-[160px] sm:w-[380px] sm:px-12">
                  <div className="absolute -right-20 -top-20 h-40 w-40 rounded-full bg-blue-500/20 blur-3xl transition-all duration-500 group-hover:bg-blue-500/30" />
                  <div className="absolute -bottom-20 -left-20 h-40 w-40 rounded-full bg-pink-500/20 blur-3xl transition-all duration-500 group-hover:bg-pink-500/30" />
                  <div className="relative flex items-center text-white">
                    <span className="mr-4 text-[3.5rem] font-extralight tracking-tighter sm:mr-6 sm:text-[4.5rem]">
                      Â£
                    </span>
                    <motion.span className="text-[3.5rem] font-extralight tracking-tighter sm:text-[4.5rem]">
                      <motion.span style={{ display: 'inline-block' }}>{price}</motion.span>
                    </motion.span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </motion.div>

        <div className="flex h-screen items-center">
          <div className="mx-auto flex w-full max-w-7xl flex-col items-center gap-8 px-4 sm:gap-4 sm:p-8 md:w-4/5 md:flex-row md:p-20">
            <img
              src={PRODUCT_1_IMAGE.src}
              alt="Strategic shopping demonstration"
              className="w-[250px] rounded-xl object-cover sm:w-[300px] md:h-[400px] md:w-[400px]"
            />
            <div className="px-4 text-center sm:px-0 md:text-left">
              <h4 className="text-xl font-semibold uppercase tracking-wide sm:text-2xl">
                Make Your Move
              </h4>
              <p className="mb-4 mt-2 w-full text-sm leading-relaxed text-primary-neutral/80 sm:text-base md:mb-6 md:w-4/5">
                Choose your moment - buy now at current pricing or wait for further reductions.
                Strategic patience may yield better prices, but popular items won't wait forever.
              </p>
              <StyledLink to="/how-it-works#step-three" />
            </div>
          </div>
        </div>
      </section>
    </div>
  );
};

export default HowItWorks;
