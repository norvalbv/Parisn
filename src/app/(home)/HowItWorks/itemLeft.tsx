import React from 'react';

const itemLeft = () => {
  return <div>itemLeft</div>;
};

export default itemLeft;
