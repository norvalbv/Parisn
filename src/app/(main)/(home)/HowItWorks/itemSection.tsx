import { cn } from '@/lib/utils/cn';
import Header from '@/src/components/Header';
import StyledLink from '@/src/components/StyledLink';
import { ReactElement } from 'react';

type ItemSectionProps = {
  imageSrc: string;
  imageAlt: string;
  title: string;
  description: string;
  isReversed?: boolean;
  href?: string;
};

const ItemSection = ({
  imageSrc,
  imageAlt,
  title,
  description,
  isReversed = false,
  href,
}: ItemSectionProps): ReactElement => {
  return (
    <div className="flex h-screen items-center">
      <div className="mx-auto flex w-full max-w-7xl flex-col items-center gap-8 px-4 sm:gap-4 sm:p-8 md:w-4/5 md:flex-row md:p-20">
        <div className={cn('flex flex-1 justify-center', isReversed && 'md:order-2')}>
          <img
            src={imageSrc}
            alt={imageAlt}
            className="w-[15.625rem] rounded-xl object-cover sm:w-[18.75rem] md:h-[25rem] md:w-[25rem]"
          />
        </div>
        <div
          className={cn(
            'flex-1 px-4 text-center sm:px-0 md:text-left',
            isReversed ? 'md:order-1' : ''
          )}
        >
          <Header title={title} />
          <p className="mb-4 mt-2 w-full md:mb-6">{description}</p>
          {href && <StyledLink href={href} />}
        </div>
      </div>
    </div>
  );
};

export default ItemSection;
