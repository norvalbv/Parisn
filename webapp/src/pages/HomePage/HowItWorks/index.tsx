import React, { ReactElement } from 'react';
import StyledLink from 'components/StyledLink/intex';
import Radial from 'components/SVG/Design';
import Phone from '../../../TempSVGImages/phone';

const HowItWorks = (): ReactElement => {
  return (
    <div className="my-10 px-20 relative">
      <div className="flex w-4/5 gap-4 items-center">
        <Phone />
        <div>
          <span>1st Choose Your Item</span>
          <p className="mb-6">
            Handpick from a carefully curated collection of exclusive brands and limited edition
            products.
          </p>
          <StyledLink to="/" />
        </div>
      </div>
      <Radial />
      <div className="flex flex-row-reverse w-4/5 gap-4 items-center relative">
        <div className="text-[32px]">
          <div className="opacity-40 ml-14">
            <span className="mr-8 text-[44px] font-normal">£</span>1,000.00
          </div>
          <div className="bg-secondary-dark rounded-3xl border border-neutral my-10 py-6 pl-10 w-96">
            <span className="mr-8 text-[44px] font-normal">£</span>500.00
          </div>
          <div className="opacity-40 ml-14">
            <span className="mr-8 text-[44px] font-normal">£</span>200.00
          </div>
        </div>
        <div>
          <span>2nd Set Your Price</span>
          <p className="mb-6">
            Set your desired price point, empowering you to take control of your shopping
            experience.
          </p>
          <StyledLink to="/" />
        </div>
      </div>
      <div className="flex w-4/5 gap-4 items-center">
        <Phone />
        <div>
          <span>3rd Win Or Wait</span>
          <p className="mb-6">
            Secure your purchase at your chosen price or hold out for a potentially better deal as
            prices continue to fall.
          </p>
          <StyledLink to="/" />
        </div>
      </div>
    </div>
  );
};

export default HowItWorks;
