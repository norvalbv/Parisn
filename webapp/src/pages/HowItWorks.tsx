import React from 'react';

const HowItWorks = () => {
  return (
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Accusantium fugiat vel doloribus
      nesciunt! Non odit reprehenderit placeat sint tempore fugit animi nemo porro quod ipsa, nisi
      provident quibusdam ipsum a ratione? Assumenda perspiciatis officia sed, hic vero maxime, vel
      dolor facilis asperiores tempora ratione ipsam fuga numquam aut deserunt odio!
    </p>
  );
};

export default HowItWorks;
