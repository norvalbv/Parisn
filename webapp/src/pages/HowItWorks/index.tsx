import CardWrapper from 'components/CardWrapper';
import React, { ReactElement } from 'react';

const HowItWorks = (): ReactElement => {
  return (
    <CardWrapper cardType="centered" className="w-1/2">
      <p className="text-center">
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Accusantium fugiat vel doloribus
        nesciunt! Non odit reprehenderit placeat sint tempore fugit animi nemo porro quod ipsa, nisi
        provident quibusdam ipsum a ratione? Assumenda perspiciatis officia sed, hic vero maxime,
        vel dolor facilis asperiores tempora ratione ipsam fuga numquam aut deserunt odio!
      </p>
    </CardWrapper>
  );
};

export default HowItWorks;
