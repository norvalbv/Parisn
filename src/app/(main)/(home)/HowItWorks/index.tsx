'use client';

import PRODUCT_1_IMAGE from '@/public/images/profile.jpg';
import Header from '@/src/components/Header';
import PriceDisplay from '@/src/components/PriceDisplay';
import { motion, useScroll, useTransform } from 'framer-motion';
import React, { ReactElement, useRef } from 'react';
import ItemSection from './itemSection';
import WavyBackground from './Waves';
import { COMPANY_NAME } from '@/src/constants';

const MAX_PRICE = 1000;
const MIN_PRICE = 0;

const HowItWorks = (): ReactElement => {
  const containerRef = useRef<HTMLDivElement>(null);
  const priceRef = useRef<HTMLDivElement>(null);

  const { scrollYProgress } = useScroll({
    target: priceRef,
    offset: ['start end', 'end start'],
  });

  const rawPrice = useTransform(scrollYProgress, [0.2, 0.8], [MAX_PRICE, MIN_PRICE]);
  const price = useTransform(rawPrice, (value: number) => value.toFixed(2));

  return (
    <div className="relative h-[280vh] sm:h-[200rem] 2xl:h-[180rem]">
      <div className="sticky top-0 h-screen w-full overflow-hidden">
        <WavyBackground
          colors={[
            'rgba(50, 117, 248, 0.15)',
            'rgba(255, 192, 203, 0.15)',
            'rgba(255, 255, 255, 0.1)',
          ]}
          backgroundFill="transparent"
          speed="fast"
          waveOpacity={0.3}
        />
      </div>
      <section ref={containerRef} className="absolute inset-0">
        <ItemSection
          imageSrc={PRODUCT_1_IMAGE.src}
          imageAlt="Luxury product showcase"
          title="Discover Your Perfect Piece"
          description="Explore our curated collection of high quality products, uniquely yours."
        />
        <motion.div
          ref={priceRef}
          className="min-h-svh sm:min-h-[100rem] top-0 relative"
          initial={{ opacity: 0, y: -20 }}
          animate={{ opacity: 1, y: 0 }}
          exit={{ opacity: 0, y: 20 }}
          transition={{ duration: 0.5 }}
        >
          <div className="sticky top-1/3 transform w-full">
            <div className="mx-auto flex w-full max-w-7xl flex-col items-center justify-between gap-8 p-4 sm:gap-16 sm:p-8 md:flex-row md:p-20">
              <div className="px-4 text-center sm:px-0 md:text-left">
                <Header title="Experience Dynamic Pricing" />
                <p className="mb-4 mt-2 w-full md:mb-6">
                  Watch prices decrease in real time across all our products. Each product is only
                  available for a short duration and will always become free if not sold.
                </p>
              </div>

              <div className="relative flex w-full justify-center md:w-2/3 md:justify-end">
                <PriceDisplay price={price} />
              </div>
            </div>
          </div>
        </motion.div>
        <ItemSection
          imageSrc={PRODUCT_1_IMAGE.src}
          imageAlt="Strategic shopping demonstration"
          title="Make Your Move"
          description="Choose your moment. Buy now at current pricing or wait for further reductions. Strategic patience may yield better prices, but popular items won't wait forever."
          isReversed
        />
      </section>
    </div>
  );
};

export default HowItWorks;
