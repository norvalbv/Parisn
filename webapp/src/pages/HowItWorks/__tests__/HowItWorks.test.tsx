import React from 'react';
import renderer from 'react-test-renderer';

import HowItWorks from '..';

describe('<HowItWorks />', () => {
  test('HowItWorks props with mandatory props', () => {
    const tree = renderer.create(<HowItWorks />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
