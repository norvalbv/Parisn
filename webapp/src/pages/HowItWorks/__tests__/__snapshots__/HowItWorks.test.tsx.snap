// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HowItWorks /> HowItWorks props with mandatory props 1`] = `
<div
  className="h-screen w-full grid place-items-center bg-primary-dark text-primary-light"
>
  <div
    className=" w-1/2"
  >
    <p
      className="text-center"
    >
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Accusantium fugiat vel doloribus nesciunt! Non odit reprehenderit placeat sint tempore fugit animi nemo porro quod ipsa, nisi provident quibusdam ipsum a ratione? Assumenda perspiciatis officia sed, hic vero maxime, vel dolor facilis asperiores tempora ratione ipsam fuga numquam aut deserunt odio!
    </p>
  </div>
</div>
`;
